lib/ansible/plugins/shell/__init__.py

```diff
@@ -101,31 +101,23 @@ def path_has_trailing_slash(self, path):
     def chmod(self, paths, mode):
         cmd = ['chmod', mode]
         cmd.extend(paths)
-        cmd = [shlex.quote(c) for c in cmd]
-
-        return ' '.join(cmd)
+        return shlex.join(cmd)
```

Comment:
semi-related: i was thinking, each shell plugin should provide a join and quote method, in this case it could just be alias to shlex ones.

Reply:
There is a quote method, would be easy to add join, and ensure we use them

Reply:
Done